Libraries/LibJS/Lexer.cpp

```diff
@@ -327,10 +327,12 @@ void Lexer::consume()
                 type = "LINE FEED"sv;
             else if (m_current_code_unit == '\r')
                 type = "CARRIAGE RETURN"sv;
-            else if (m_source.code_unit_at(m_position + 1) == LINE_SEPARATOR)
+            else if (current_code_point() == LINE_SEPARATOR)
                 type = "LINE SEPARATOR"sv;
-            else
+            else if (current_code_point() == PARAGRAPH_SEPARATOR)
                 type = "PARAGRAPH SEPARATOR"sv;
+            else
+                VERIFY_NOT_REACHED();
```

Comment:
These are non-surrogate UTF-16 code units, so these can both just reference `m_current_code_unit`:
```c++
if (m_current_code_unit == '\n')
    type = "LINE FEED"sv;
else if (m_current_code_unit == '\r')
    type = "CARRIAGE RETURN"sv;
else if (m_current_code_unit == LINE_SEPARATOR)
    type = "LINE SEPARATOR"sv;
else if (m_current_code_unit == LINE_SEPARATOR)
    type = "PARAGRAPH SEPARATOR"sv;
else
    VERIFY_NOT_REACHED();
```

Reply:
I've made the changes to the code as you've indicated.  And updated the commit message as best I could.

Reply:
I also tested it again, and numbers are like so

```
     15 Found a line terminator: CARRIAGE RETURN
 805690 Found a line terminator: LINE FEED
     14 Found a line terminator: LINE SEPARATOR
     14 Found a line terminator: PARAGRAPH SEPARATOR
```